core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/console-reporter.ts

```diff
@@ -0,0 +1,191 @@
+  JSHandle,
+} from 'puppeteer';
+
+import escapeRegExp from 'lodash/escapeRegExp';
+
+const HOST_URL = 'http://localhost:8181';
+
+interface ConsoleMessage {
+  type: ConsoleMessageType;
+  text: string;
+  url: string;
+}
+
+const CONSOLE_ERRORS_TO_IGNORE = [
+  // These "localhost:9099" are errors related to communicating with the
+  // Firebase emulator, which would never occur in production, so we just ignore
+  // them.
+  escapeRegExp(
+    'http://localhost:9099/www.googleapis.com/identitytoolkit/v3/' +
+      'relyingparty/getAccountInfo?key=fake-api-key'
+  ),
+  escapeRegExp(
+    'http://localhost:9099/www.googleapis.com/identitytoolkit/v3/' +
+      'relyingparty/verifyPassword?key=fake-api-key'
+  ),
+  // This error covers the case when the PencilCode site uses an
+  // invalid SSL certificate (which can happen when it expires).
+  // In such cases, we ignore the error since it is out of our control.
+  escapeRegExp(
+    'https://pencilcode.net/lib/pencilcodeembed.js - Failed to ' +
+      'load resource: net::ERR_CERT_DATE_INVALID'
+  ),
+  // These errors are related to the gtag script that is used to track events.
+  // They are of the form "Failed to load resource: the server responded
+  // with a status of 405", this happens when the HTTP method used for a
+  // network call is refused by the server. The network call is triggered
+  // automatically by the gtag script, so we have no control over it. The 405
+  // error was observed on other websites (e.g. https://edu.google.com/) that
+  // use gtag and it does not affect the user experience in any way.
+  // Considering these reasons, the error may be ignored.
+  new RegExp(
+    'https://www.googletagmanager.com/a.* Failed to load resource: ' +
+      'the server responded with a status of 405 ()',
+    'g'
+  ),
+];
+
+const CONSOLE_ERRORS_TO_FIX = [
+  // TODO(#19746): Development console error "Uncaught in Promise" on signup.
+  new RegExp(
+    'Uncaught \\(in promise\\).*learner_groups_feature_status_handler'
+  ),
+  // TODO(#19687): 404 (Not Found) for favicon on home page.
+  escapeRegExp(
+    'http://localhost:8181/dist/oppia-angular-prod/favicon.ico Failed ' +
+      'to load resource: the server responded with a status of 404 (Not Found)'
+  ),
+  // TODO(#19733): 404 (Not Found) for resources used in midi-js.
+  escapeRegExp(
+    'http://localhost:8181/dist/oppia-angular/midi/examples/soundfont/acoustic' +
+      '_grand_piano-ogg.js Failed to load resource: the server responded with a ' +
+      'status of 404 (Not Found)'
+  ),
+];
+
+export class ConsoleReporter {
+  private static consoleMessages: ConsoleMessage[] = [];
+  private static consoleErrorsToIgnore: (RegExp | string)[] = [];
+
+  /**
+   * This function initializes tracking for the browser's console messages.
+   */
+  public static trackConsoleMessagesInBrowser(browser: Browser): void {
+    browser.on('targetcreated', async (target: Target) => {
+      if (target.type() === 'page') {
```

Comment:
Can we do something like this instead:
```ts
    if (target.type() !== 'page') {
      return;
    }

    ...
```
This will reduce nesting and shorten indents, which makes the code and logic easier to read.

Reply:
Done